src/zenml/zen_server/auth.py

```diff
@@ -207,3 +210,39 @@ def authentication_provider() -> Callable[..., AuthContext]:
 
 
 authorize = authentication_provider()
+
+
+def user_has_write_permissions(auth_context: AuthContext = Depends(authorize)):
+    # Get the corresponding ids for the authorized roles
+    required_permission = PermissionType.WRITE.value
+
+    current_user = auth_context.user
+    # Get all global role assignments for this user
+    roles = zen_store().list_role_assignments(
+        user_name_or_id=current_user.id,
+        project_name_or_id=None
+    )
+
+    for role in roles:
+        for permission in zen_store().get_role(role.role).permissions:
+            if permission == required_permission:
+                return True
+    raise NotAuthorizedError("Unauthorized")
+
+
+def user_has_read_permissions(auth_context: AuthContext = Depends(authorize)):
+    # Get the corresponding ids for the authorized roles
+    required_permission = PermissionType.READ.value
+
+    current_user = auth_context.user
+    # Get all global role assignments for this user
+    roles = zen_store().list_role_assignments(
+        user_name_or_id=current_user.id,
+        project_name_or_id=None
+    )
+
+    for role in roles:
+        for permission in zen_store().get_role(role.role).permissions:
+            if permission == required_permission:
+                return True
+    raise NotAuthorizedError("Unauthorized")
```

Comment:
this is mostly duplicate code from above, I'd suggest define a helper method `user_has_permission(auth_context, permission)` and then just call `user_has_permission(auth_context, PermissionType.READ.value` here

Reply:
I moved away from this so no longer necessary